modules/common/security/networking.nix

```diff
@@ -0,0 +1,201 @@
+# Copyright 2024-2025 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  ...
+}: let
+  cfg = config.ghaf.security.sysctl.network;
+in {
+  ## Options to enable IP security
+  options.ghaf.security.sysctl.network = {
+    disable-all = lib.mkOption {
+      description = ''
+        Disable all sysctl network security configuration.
+      '';
+      type = lib.types.bool;
+      default = false;
+    };
+
+    disable-ipv6 = lib.mkOption {
+      description = ''
+        Disables Internet Protocol v6.
+        It is recommended to reduce attack surface by disabling IPv6 if it is not being used.
+        References:
+        https://www.tenable.com/audits/items/CIS_AlmaLinux_OS_8_Server_v1.0.0_L2.audit:a532993e3ac29c19c0d491463a881303
+        https://linux-audit.com/linux-security-guide-for-hardening-ipv6/
+      '';
+      type = lib.types.bool;
+      default = true;
```

Comment:
Should this be default to true?

Reply:
OK, I set default to `false`. In Ghaf by default IPv6 is disabled.
https://github.com/tiiuae/ghaf/blob/main/modules/microvm/virtualization/microvm/common/vm-networking.nix#L24C23-L24C24

Reply:
OK, I set default to `false`. In Ghaf by default IPv6 is disabled.
https://github.com/tiiuae/ghaf/blob/main/modules/microvm/virtualization/microvm/common/vm-networking.nix#L24C23-L24C24

Reply:
If we enable IPv6 we should include hardening settings as well. I agree we should support IPv6, but enabling it indiscriminately in all VMs and interfaces may not be the best strategy with our current setup, and afaik enabling it for specific interfaces is still possible with these settings